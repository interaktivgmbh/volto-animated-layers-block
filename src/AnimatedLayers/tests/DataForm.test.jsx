import React from 'react';
import AnimatedLayersDataForm from '../DataForm';
import AnimatedLayersSchema from '../Schema';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import configureStore from 'redux-mock-store';
//noinspection JSUnresolvedFunction, NpmUsedModulesInstalled
import config from '@plone/volto/registry';
import { Provider } from 'react-intl-redux';

const mockStore = configureStore();

//noinspection JSUnresolvedFunction
beforeAll(() => {
  config.blocks.blocksConfig = {
    ...config.blocks.blocksConfig,

    animatedlayers: {
      id: 'animatedlayers',
    },
  };
});

describe('AnimatedLayersDataForm', () => {
  it('renders AnimatedLayersDataForm component', () => {
    const store = mockStore({
      intl: {
        locale: 'en',
        messages: {},
      },
    });

    const testSchema = AnimatedLayersSchema({});

    const data = {
      '@type': 'animatedlayers',
      layers: [],
    };
    const component = renderer.create(
      <Provider store={store}>
        <AnimatedLayersDataForm key={'1234'} schema={testSchema} data={data} block={'1234'} onChangeBlock={null} />
      </Provider>,
    );
    const json = component.toJSON();
    //noinspection JSUnresolvedFunction
    expect(json).toMatchSnapshot();
  });
});
