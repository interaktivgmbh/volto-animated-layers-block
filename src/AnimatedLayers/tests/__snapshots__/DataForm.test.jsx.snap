// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimatedLayersDataForm renders AnimatedLayersDataForm component 1`] = `
<div
  className="ui form"
>
  <div
    id="blockform-fieldset-default"
  >
    <div
      className="ui segment form attached"
    >
      <div
        className="mocked-default-widget"
        id="mocked-field-title"
      >
        Title
         - 
        Title of the Block (left side).
      </div>
      <div
        className="mocked-default-widget"
        id="mocked-field-description"
      >
        Description
         - 
        Description of the Block (left side).
      </div>
    </div>
  </div>
</div>
`;
